### DISCUSSION TITLE
Query string with array parameters bug

### DISCUSSION THREAD

**Discussion by Lupennat** [livewire/livewire#4473] [7. Bugs (archived)]
I couldn't find any documentation about it, so I assume that using array type properties in the querystring is supported.

However, using an array type property, I noticed that there is a comparison issue in the browserSync.js file.

```js
if ('path' in effects && effects.path !== window.location.href) 
```

When query string parameters in the browser are defined with this syntax `url?filters[search]=search` Php generates effects.path with a different syntax `url?filters%5Bsearch%5D=search`

In addition to this, I have also verified that the generation of the parameters is not always correct in the SupportBrowserHistory.php.
assuming that
`$queryParams = [ 'filters' => [search" => null, "status" => null ] ];`

`$queryParams->isEmpty()` is false, but the generated query is an empty string and an unnecessary '?' is appended to effects.path.

I tried locally a solution that is to be verified that solves both problems by modifying the function like this

```php
  protected function stringifyQueryParams($queryParams)
    {
        if ($queryParams->isEmpty()) {
            return '';
        }

        $params = urldecode(http_build_query($queryParams->toArray(), '', '&', PHP_QUERY_RFC1738));

        return $params ? '?' . $params : '';
    }
```

I haven't checked compatibility with all browsers at all.

**Comment by joshhanley** [comment]
@Lupennat thanks for reporting! If you have a solution for this, any chance you could submit a PR for it? Thanks!

**Reply by Lupennat** [reply]
@joshhanley yes, of course, I can make a pull request, can you tell me if there is a Contribution Guide to refer to?

**Reply by joshhanley** [reply]
@Lupennat thanks! Yep check it out here https://laravel-livewire.com/docs/2.x/contribution-guide

**Reply by Lupennat** [reply]
@joshhanley pull request submitted https://github.com/livewire/livewire/pull/4568

**Reply by joshhanley** [reply]
@Lupennat thanks!

**Reply by Lupennat** [reply]
@joshhanley do you have any idea how long it will take for the pull request to be approved and the tag to be created?

**Reply by joshhanley** [reply]
@Lupennat Caleb is currently preparing for Laracon, so hopefully once that is over he should have some time to go through them 🙂
